specifyweb/specify/views.py

```diff
@@ -555,5 +583,85 @@ def record_merge(request: http.HttpRequest, model_name: str, old_model_id: int,
     # if model_name.lower() != 'Agent'.lower():
     #     return http.HttpResponseForbidden(f'Record Merging for the {model_name} table is not yet been fully tested.')
 
-    response = record_merge_fx(model_name, old_model_id, new_model_id)
+    response = record_merge_fx(model_name, [old_model_id], new_model_id)
+    return response
+
+@openapi(schema={
+    'post': {
+        "requestBody": {
+            "required": True,
+            "description": "Rplace a list of old records with a new record.",
+            "content": {
+                "application/json": {
+                    "schema": {
+                        "type": "object",
+                        "description": "The request body.",
+                        "properties": {
+                            "model_name": {
+                                "type": "string",
+                                "description": "The name of the table that is to be merged."
+                            },
+                            "new_model_id": {
+                                "type": "integer",
+                                "description": "The new ID value of the model that is replacing the old one."
+                            },
+                            "old_record_ids": {
+                                "type": "array",
+                                "items": {
+                                    "type": "integer"
+                                },
+                                "description": "The old record IDs."
+                            },
+                            "new_record_data": {
+                                "type": "object",
+                                "description": "The new record data."
+                            }
+                        },
+                        'required': ['model_name', 'new_model_id', 'collection_id', 'old_record_ids', 'new_record_data'],
+                        'additionalProperties': False
+                    }
+                }
+            }
+        },
+        "responses": {
+            "204": {"description": "Success",},
+            "404": {"description": "The ID specified does not exist."},
+            "405": {"description": "A database rule was broken."}
+        }
+    },
+})
+
+@login_maybe_required
+@require_POST
+def record_merge_extended(request: http.HttpRequest, model_name: str, new_model_id: int) -> http.HttpResponse:
```

Comment:
I am not too familiar with django/python/sqlalcheme, but it this code being run as a transaction? i.e., if some records get deleted, but then it falls mid-way, (potentially because of missing permissions), do those records get un-deleted?

additionally, if those records have attachments, does the back-end send requests to the asset server to delete the attachments? (it might have triggers doing that) that might cause issues if request doesn't go though fully

Reply:
ya, I made sure to research it.  The `@transaction.atomic` decorator encapsulates a database actions into one ACID transactions.  Even in the case of recursively calling the function, creating nested transactions, either all the actions occur, or none of them.  I'm not sure about the attachments, it's not something I can unit test easily.  We should make sure to test that testcase in the integration testing.  I'll look to see if I can find any triggers for deleting attachments, but I don't think it would cause a trigger to occur until the full atomic transaction is completed.